troi/core.py

```diff
@@ -1,72 +1,65 @@
 #!/usr/bin/env python3
 
 import sys
+from typing import Dict
 
 import click
 
 import troi
 import troi.playlist
 import troi.utils
-
-
-def generate_playlist(patch, debug=False, echo=True, save=False, token=None, upload=False, args=None,
-                      created_for=None, name="", desc="", min_recordings=10):
+from troi.patch import Patch
+
+default_patch_args = dict(
+    debug=False,
+    echo=True,
+    save=False,
+    token=None,
+    upload=False,
+    args=None,
+    created_for=None,
+    name=None,
+    desc=None,
+    min_recordings=10
+)
+
+
+def generate_playlist(patch: Patch, pipeline_args: Dict, supplied_patch_args: Dict):
```

Comment:
I had initially made this change because I used a TypedDict for documenting the params and it felt easier but then I realised we may be unable to use TypedDict if we want to support Py 3.6 and 3.7 so fine to revert to separate args I guess if we want.

Reply:
Discussed in IRC. will open a separate PR to use TypedDict and baseline at python 3.8